### DISCUSSION TITLE
Question: Easiest way to add Mastodon verification to social link

### DISCUSSION THREAD

**Discussion by mycoporium** [getpelican/pelican#3254] [Q&A] [answered]
In order to verify a Mastodon user, you simply add `rel="me"` to your website's profile link. If using a Pelican blog, this isn't so straightforward.

I was considering following the [Injector Plugin](https://github.com/pelican-plugins/injector/tree/main) model, and using BS4 to modify the HTML output before final write. Another thought was to make a customized version of the default template, which presumably would use Jinja2 logic to insert the `rel` attribute at the correct social link.

Which of these two paths is the most in line with the preferred way of doing things? Is there another method that might be better?

Bonus points: It doesn't look like Mastodon is a known social link, and my site isn't displaying its icon. Is there a way I can add it? I assume this is done in CSS because I didn't see the images for YouTube, GitHub, etc, rendered in the HTML.

Thanks!

**Comment by Epy** [comment]
Hello
It seem pretty straightforward to me: In my theme I added attributes for icons and links:

In one of the template files: (bottom of navigation in my case)
```
{% if BADGES %}
    {% for name, imgsrc, link, attr  in BADGES %}
        <a href="{{ link }}" title="{{ name }}" {% if attr != '' %}{{ attr }}{% endif %}><img src="{{ SITEURL }}/{{ THEME_STATIC_DIR }}/{{ imgsrc }}" alt="{{ name }}"></a>
    {% endfor %}
{% endif %}
```
In pelicanconf:
```
BADGES = (('CC BY-SA', 'img/gadgets/cc-by-sa.png', 'http://creativecommons.org/licenses/by-sa/4.0/', 'alt="Creative Commons By SA" '),
    ('Pelican', 'img/gadgets/pelican.png', 'http://getpelican.com/', 'alt="Pelican icon" '),
 ('Mastodon', '<mastodon image path>','<mastodon profile url>','rel=me'),
    )
```


[I removed it so I show you placeholders, but I did it like that, simple enough. It may be simplified again, maybe]

**Reply by mycoporium** [reply]
Thanks for the example! It does seem pretty straightforward, but I'm not using a custom theme. I'd like to try keeping that way, but I just might have to do something like this. Thanks again.

**Comment by justinmayer** [comment, marked as answer]
The short answer to both questions is to either (1) submit a PR that modifies the upstream theme (presumably the default `notmyidea` theme?), or (2) copy that theme and modify it locally.

Longer answer for the first question: I think perhaps a PR adding `rel="me"` to [the relevant line in the theme template](https://github.com/getpelican/pelican/blob/d9b2bc3a4ea9f11aea58180aaaea69cc95db1f8c/pelican/themes/notmyidea/templates/base.html#L65) might be one way? That is:

```html
<li><a rel="me" href="{{ link }}">{{ name }}</a></li>
```

Longer answer for the second question: I believe all the social icons were recently removed from the default theme due to concerns about provenance, copyright, etc. If someone were to submit a PR adding an "unencumbered" icon for Mastodon, I imagine that would be gladly received and merged. 😊

**Reply by mycoporium** [reply]
Thank you! This is great info. I'm currently hacking a custom plugin together, but I like the idea of figuring out how to do an "unencumbered icon" as you say. I just might add that PR. Cheers!
